src/app/modules/static-map-image/static-map-image.component.ts

```diff
@@ -0,0 +1,445 @@
+import { NgDestoryBase } from 'src/app/core/helpers/observable-helper';
+import { Feature, Polygon } from '@turf/turf';
+import { END_ICON, START_ICON } from '../map-image/map-image.component';
+import { LoggingService } from '../shared/services/logging/logging.service';
+
+interface TileProps {
+  src: SafeUrl;
+  top: string;
+  left: string;
+}
+
+interface Graphic {
+  id: string;
+  svg: SafeHtml;
+  style: { [styleDesc: string]: number };
+}
+
+const trackByImgProps: TrackByFunction<TileProps> = (index, item) => item.src;
+const trackByGraphic: TrackByFunction<Graphic> = (index, item) => item.id;
+
+// We only have map services with 256px tiles at the moment.
+const TILE_SIZE = 256;
+const PADDING = 15;
+
+interface PositionToPlot {
+  pos: ImageLocation['latLng'];
+  type: 'start' | 'stop' | 'damage' | 'obs';
+  px?: { x: number; y: number };
+}
+
+interface LatLngBounds {
+  minLng: number;
+  minLat: number;
+  maxLng: number;
+  maxLat: number;
+}
+
+// n, s, e, w in pixels from top left of world
+interface MercatorBounds {
+  n: number;
+  s: number;
+  e: number;
+  w: number;
+  zoom: number;
+}
+
+const createGeojsonBounds = ({ minLng, minLat, maxLng, maxLat }: LatLngBounds): Feature<Polygon> => ({
+  type: 'Feature',
+  properties: {},
+  bbox: [minLng, minLat, maxLng, maxLat],
+  geometry: {
+    type: 'Polygon',
+    coordinates: [
+      [
+        [minLng, minLat],
+        [maxLng, minLat],
+        [maxLng, maxLat],
+        [minLng, maxLat],
+        [minLng, minLat],
+      ],
+    ],
+  },
+});
+
+@Component({
+  selector: 'app-static-map-image',
+  templateUrl: './static-map-image.component.html',
+  styleUrls: ['./static-map-image.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: MapLayersService,
+      useClass: isPlatform('hybrid') ? OfflineCapableMapLayersService : MapLayersService,
+    },
+  ],
+})
+export class StaticMapImageComponent extends NgDestoryBase implements AfterViewInit {
+  @Input() location: ImageLocation;
+  @Input() allowZoom: boolean;
+
+  @ViewChild('container', { static: true })
+  container: ElementRef<HTMLDivElement>;
+
+  @HostListener('window:resize', ['$event'])
+  onResize() {
+    this.componentCreatedOrResized.next();
+  }
+
+  tiles: TileProps[] = null;
+  graphics: Graphic[] = [];
+
+  private componentCreatedOrResized = new Subject<void>();
+  private size = new ReplaySubject<{ w: number; h: number }>(1);
+  size$ = this.size.pipe(
+    // NB! The debounceTime here must be lower than the interval time
+    // specified in startSizeFinder, if not this may not emit.
+    debounceTime(50),
+    skipWhile(({ w, h }) => w === 0 || h === 0),
+    map(({ w, h }) => ({
+      w: w.toFixed(0),
+      h: h.toFixed(0),
+    })),
+    distinctUntilChanged((prev, curr) => prev.h === curr.h && prev.w === curr.w),
+    map(({ w, h }) => ({ w: +w, h: +h })),
```

Comment:
Jeg forstår ikke hvorfor vi må gjøre alt dette med størrelsen. For meg ser det ut som en veldig rar måte å runde av bredden og høyden på, men jeg regner med at det er en grunn til det:) Men hadde vært supert med en kommentar om hvorfor du gjør det.